lampod/src/jsonrpc/channels.rs

```diff
@@ -8,3 +16,86 @@ pub fn json_list_channels(ctx: &LampoDeamon, request: &json::Value) -> Result<js
     let resp = ctx.channel_manager().list_channel();
     Ok(json::to_value(resp)?)
 }
+
+pub fn json_close_channel(ctx: &LampoDeamon, request: &json::Value) -> Result<json::Value, Error> {
+    log::info!("call for `closechannel` with request {:?}", request);
+    let mut request: request::CloseChannel = json::from_value(request.clone())?;
+    let events = ctx.handler().events();
+    let res;
+    // This gives all the channels with associated peer
+    let channels: response::Channels = ctx
+        .handler()
+        .call(
+            "channels",
+            json::json!({
+            "peer_id": request.node_id,
+            }),
+        )
+        .map_err(|err| {
+            Error::Rpc(RpcError {
+                code: -1,
+                message: format!("{err}"),
+                data: None,
+            })
+        })?;
+    if channels.channels.len() > 1 {
+        // check the channel_id if it is not none, if it is return an error
+        // and if it is not none then we need to have the channel_id that needs to be shut
+        if request.channel_id.is_none() {
+            return Err(Error::Rpc(RpcError {
+                code: -1,
+                message: format!("Channels > 1, provide `channel_id`"),
+                data: None,
+            }));
+        } else {
+            res = ctx.channel_manager().close_channel(request.clone());
+        };
+    } else if !channels.channels.is_empty() {
+        // This is the case where channel with the given node_id = 1
+        // SAFETY: it is safe to unwrap because the channels is not empty
+        let channel = channels.channels.first().unwrap();
+        request.channel_id = Some(channel.channel_id.clone());
+        res = ctx.channel_manager().close_channel(request);
+    } else {
+        // No channels with the given peer.
+        return Err(Error::Rpc(RpcError {
+            code: -1,
+            message: format!("No channels with associated peer"),
+            data: None,
+        }));
+    };
+    let (message, channel_id, node_id, funding_utxo) = loop {
+        let event = events
+            .recv_timeout(std::time::Duration::from_secs(30))
+            .map_err(|err| {
+                Error::Rpc(RpcError {
+                    code: -1,
+                    message: format!("{err}"),
+                    data: None,
+                })
+            })?;
+        if let Event::Lightning(LightningEvent::CloseChannelEvent {
+            message,
+            channel_id,
+            counterparty_node_id,
+            funding_utxo,
+        }) = event
+        {
+            break (message, channel_id, counterparty_node_id, funding_utxo);
+        }
+    };
+    let resp = match res {
```

Comment:
```suggestion
    match ctx.channel_manager().close_channel(request) {
```

Reply:
I think that using the resp in this was should be an antipattern in rust

Reply:
@Harshit933 ok you can call the method when it is necessary, but avoid to do the trick with the `let ret` because I do not like

Reply:
Sure.